odoo/addons/base/models/res_partner.py

```diff
@@ -696,6 +698,11 @@ def _fields_sync(self, values):
         # 2. To DOWNSTREAM: sync children
         self._children_sync(values)
 
+        # 3. To UPSTREAM: force parent update
+        if has_address_field_set and self.parent_id and self.type == 'contact':
```

Comment:
The line below is commented: either explain why or remove it.
You don't need `has_address_field_set` because `update_address` already handles that.
You shouldn't update parents that are companies either (commercial field sync is only from a company to its employees) and this function needs to handle recordsets with multiple values.

```suggestion
        if parents := (self.filtered(lambda p: p.type == 'contact').commercial_partner_id - self):
```

Reply:
As specs are being discussed live, this is going to evolve :D .

Reply:
Then the PR could be set to draft :)